src/azul/lambdas.py

```diff
@@ -178,3 +202,18 @@ def reset_lambda_roles(self):
                             time.sleep(1)
                         else:
                             break
+
+    def delete_stale_function_versions(self):
+        """
+        Delete all but the latest published version of every AWS Lambda function
```

Comment:
```suggestion
        Delete all but the latest published version of every AWS Lambda function
```

Can there be more than one published version? If not, "the latest" qualifier seems redundant.

Reply:
> Can there be more than one published version?

Yes, however I was using the term `published` incorrectly. In AWS terminology a `published` version is a copy of the unpublished ($LATEST) version. There can be many published versions (each with a unique version number), but only one unpublished version.

https://docs.aws.amazon.com/lambda/latest/dg/configuration-versions.html

In my latest fixup commit I've changed the name and docstring of this method accordingly.

Reply:
Then I believe either the deletion logic is wrong or the `publish` TF functionality doesn't work. One of the published versions must be live so the logic below deletes it, or the latest version is live which raises the question as to why we published an immutable version.

Reply:
I might of misunderstood your original question. There can be multiple published versions of a function in the sense that AWS allows it, however with this PR there will only ever be at most one published version, along with the unpublished version.

Both the published and unpublished versions are live, however only the unpublished version is used since our code does not qualify the function ARN with a version. https://docs.aws.amazon.com/lambda/latest/dg/configuration-versions.html#versioning-versions-using

Reply:
Still don't understand it. Please raise in PL.